cfgov/mega_menu/frontend_conversion.py

```diff
@@ -0,0 +1,148 @@
+class FrontendConverter:
+    def __init__(self, menu, request=None):
+        self.menu = menu
+        self.request = request
+
+    def get_menu_items(self):
+        return [
+            self.get_menu_item(submenu.value) for submenu in self.menu.submenus
+        ]
+
+    def get_menu_item(self, submenu):
+        menu_item = {
+            'link': self.make_link({
+                'page': submenu.get('overview_page'),
+                'text': submenu.get('title'),
+            }),
+        }
+
+        columns = self.get_columns(submenu)
+        if columns:
+            menu_item['nav_groups'] = columns
+
+        # For legacy menu
+        footer = self.get_footer(submenu)
+        if footer:
+            menu_item['footer'] = footer
+
+        featured_content = self.get_featured_content(submenu)
+        if featured_content:
+            menu_item['featured_content'] = featured_content
+
+        # For new menu variations
+        featured_links = self.make_links(submenu.get('featured_links'))
+        if featured_links:
+            menu_item['featured_links'] = featured_links
+
+        other_links = self.make_links(submenu.get('other_links'))
+        if other_links:
+            menu_item['other_links'] = other_links
+
+        return menu_item
+
+    def get_columns(self, submenu):
+        columns = []
+        last_heading = None
+
+        for column in (submenu.get('columns') or []):
+            heading = column.get('heading')
+
+            columns.append({
+                'value': {
+                    'group_title': heading or last_heading,
+                    'hide_group_title': not heading,
+                    'nav_items': [
+                        self.make_link(link)
+                        for link in (column.get('links') or [])
+                    ],
+                },
+            })
+
+            last_heading = heading
+
+        return columns
+
+    def get_footer(self, submenu):
+        footer_links = []
+
+        for other_link in submenu.get('other_links') or []:
+            page = other_link.get('page')
+
+            if page:
+                link = self.make_link_for_page(page)
+                text = other_link['text']
+                url = link['external_link']
+                link_text = link['link_text']
+            else:
+                text = ''
+                url = other_link['url']
+                link_text = other_link['text']
+
+            html = '<p>'
+
+            if text:
```

Comment:
This logic is copied from [how the existing menu does this](https://github.com/cfpb/cfgov-refresh/blob/b97c47871d3658e49eba284690de0b47df34a90c/cfgov/v1/models/menu_item.py#L111-L129), and will be unnecessary if/when we move to a menu with simpler links that don't require this markup.